fetch.js

```diff
@@ -37,6 +37,25 @@
     return value
   }
 
+  function getIterator(headers, addItem) {
+    var items = []
+    headers.forEach(addItem, items)
+    var iterator = {
+      next: function() {
+        var value = items.shift()
+        return {done: value === undefined, value: value}
+      }
+    }
+
+    if (support.iterable) {
+      iterator[Symbol.iterator] = function() {
+        return iterator
+      }
+    }
+
+    return iterator
+  }
```

Comment:
It sounds like a better approach. Do you want me to update the code based on your comment?